macros/src/codegen.rs

```diff
@@ -3,11 +3,24 @@
 use crate::parser::lifetimes::Lifetimes;
 use crate::parser::ParsedStateMachine;
 use proc_macro2::Span;
-use quote::quote;
+use quote::{quote, format_ident};
 use std::vec::Vec;
-use syn::{punctuated::Punctuated, token::Paren, Type, TypeTuple};
+use syn::{punctuated::Punctuated, token::Paren, Type, TypeTuple, Ident};
 
 pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
+    let states_type_name = sm.name.as_ref()
+        .map(|i| format_ident!("{}States", i))
+        .unwrap_or_else(||Ident::new("States", Span::call_site()));
+    let events_type_name = sm.name.as_ref()
+        .map(|i| format_ident!("{}Events", i))
+        .unwrap_or_else(||Ident::new("Events", Span::call_site()));
+    let state_machine_type_name = sm.name.as_ref()
+        .map(|i| format_ident!("{}StateMachine", i))
+        .unwrap_or_else(||Ident::new("StateMachine", Span::call_site()));
+    let state_machine_context_type_name = sm.name.as_ref()
+        .map(|i| format_ident!("{}StateMachineContext", i))
+        .unwrap_or_else(||Ident::new("StateMachineContext", Span::call_site()));
```

Comment:
What about something a little simpler to reduce accidentally naming things different with vs without name?

```suggestion
    let name = sm.name.as_ref().unwrap_or("");
    let states_type_name = format_ident!("{name}States");
    let events_type_name = format_ident!("{name}Events");
    let state_machine_type_name = format_ident!("{name}StateMachine");
    let state_machine_context_type_name = format_ident!("{name}StateMachineContext");
```